gallery_dl/extractor/weasyl.py

```diff
@@ -190,17 +252,99 @@ class WeasylJournalsExtractor(WeasylExtractor):
 
     def __init__(self, match):
         WeasylExtractor.__init__(self, match)
-        self.owner_login = match.group(1)
+        self.owner_login = match[1]
 
     def items(self):
-        yield Message.Version, 1
-        yield Message.Directory, {"owner_login": self.owner_login}
-        response = self.request("{}/journals/{}".format(
-            self.root, self.owner_login
-        ))
-
-        for journal in re.finditer(r'"/journal/(\d+)/([\w-]+)"',
-                                   response.text):
-            data = self.retrieve_journal(int(journal.group(1)))
-            data["title"] = journal.group(2)
-            yield Message.Url, data["html"], data
+        yield from WeasylExtractor.items(self)
+        yield from self.scrape("journal")
+
+
+class WeasylUserExtractor(WeasylExtractor):
+    subcategory = "user"
+    pattern = BASE_PATTERN + r"~([\w-]+)"
+    test = ("https://www.weasyl.com/~tanidareal", {
+        })
+
+    def __init__(self, match):
+        WeasylExtractor.__init__(self, match)
+        self.owner_login = match[1]
+
+    def items(self):
+        yield from WeasylExtractor.items(self)
+        yield from self.submissions()
+        yield from self.scrape("character")
+        yield from self.scrape("journal")
```

Comment:
This should be done like
https://github.com/mikf/gallery-dl/blob/9a9d1924d87d48c8093264b61b515dc4feb1bc53/gallery_dl/extractor/furaffinity.py#L291-L297

(or the `user` extractors in nijie.py, newgrounds.py, hentaifoundry.py, etc)